frontend/catalyst/utils/tracing.py

```diff
@@ -26,14 +30,26 @@ class TracingContext:
     """
 
     _is_tracing = False
+    _supported_version = "0.4.13"
 
     def __enter__(self):
+        if not TracingContext.is_supported_jax_version():
+            jax_version = jax.version.__version__
+            msg = f"Attempting to trace with JAX version {jax_version}"
+            msg += f" when supported version is {TracingContext._supported_version}."
+            warnings.warn(msg)
```

Comment:
@erick-xanadu two quick comments:

- it would be good to be explicit about the the warning type here (the default `UserWarning` is probably the correct one from looking at the docs, but `RuntimeWarning` could also work https://docs.python.org/3/library/warnings.html#warning-categories)

- A good user warning is always _actionable_. That is, it should tell the user how to avoid it. I recommend something along the lines of "Attempting to QJIT a function using JAX version ..., however Catalyst currently only supports JAX version .... If you run into issues, please install the supported version of JAX."